py/htm/algorithms/apical_tiebreak_temporal_memory.py

```diff
@@ -0,0 +1,1126 @@
+# ----------------------------------------------------------------------
+# HTM Community Edition of NuPIC
+# Copyright (C) 2017, Numenta, Inc. 
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.    If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""An implementation of TemporalMemory"""
```

Comment:
- [ ] please add a note to files that are ported from htm.research that they come from there. 
- [ ] mention that the API may break (from research), you may add a note how lively/stable the code is now
- [ ] use a separate "namespace" for these. py/htm/{experimental, research?}/algorithms

Reply:
- [ ] Okay I will add the note. Should have done that from the beginning.
- [ ] I don't want to use the namespace "htmresearch" since I consider these as being usable and Numenta is no longer supporting them. 
- [ ] I cannot see these API's breaking for the same reason.
- [ ] I suggest the namespace "htm_advanced" since this is what they really are. Also this is the namespace that I used when developed the port so it will be quicker to make the change back.

Reply:
> as being usable and Numenta is no longer supporting them.

interesting point. Numenta is abandoning their whole htmresearch repo? Due to the py3 switch their trying to pull? 

>  I cannot see these API's breaking for the same reason.

ok, good idea would be to make a `htm_advanced/README.md` file and mention all these things:
- that the files are ported from htmresearch repo
- that (and why?) Numenta is no longer developing them
  - comes with API being stable
  - note, the files directly use Connections. We consider connections our internal code, so the APIs may break, but not likely. 
  - htm_advanced is fine, esp if it'd be easy for you to revert to that. Otherwise I'd vote only for "advanced", or some other names.

Reply:
> Numenta is abandoning their whole htmresearch repo? Due to the py3 switch their trying to pull?

I wouldn't say we are abandoning it. We'll port the code we need again to py3 as we need it. These implementations in Py2 are all we have, but we probably won't be running python2 runtimes anymore. So they are here until we need to port them (as @fcr just did -- thanks, btw!)

Reply:
Thanks for the clarification, Matt!